bundles/com.espressif.idf.lsp/src/com/espressif/idf/lsp/preferences/IDFClangdPreferredOptions.java

```diff
@@ -0,0 +1,94 @@
+package com.espressif.idf.lsp.preferences;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.eclipse.cdt.lsp.clangd.ClangdMetadata;
+import org.eclipse.cdt.lsp.clangd.ClangdOptions;
+import org.eclipse.core.runtime.preferences.IScopeContext;
+import org.eclipse.core.runtime.preferences.PreferenceMetadata;
+
+public class IDFClangdPreferredOptions implements ClangdOptions
+{
```

Comment:
As we already have **IDFClangdOptionsDefaults** which implements **ClangdOptionsDefaults**, can we use that instead of implementing again ClangdOptions?

Reply:
@sigmaaa Could you check this?

Reply:
Hi @kolipakakondal,

The **IDFClangdOptionsDefaults** class doesn't use metadata to store preferences. Instead, it provides default settings that can be applied universally. On the other hand, **IDFClangdPreferredOptions** retrieves stored preferences specifically from the metadata.

So both classes are in use:

- **IDFClangdOptionsDefaults** ensures there are baseline default settings.
- **IDFClangdPreferredOptions** accesses and applies preferences saved by the user or project.

Reply:
Ok, should be fine. Having two separate classes makes the code more readable, instead of handling if(metadata != null).